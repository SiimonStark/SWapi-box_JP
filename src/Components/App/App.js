import React, { Component } from 'react';
import { findRandomFilm, findPeople } from '../../helper_Fetch';
import Header from '../Header/Header';
import DisplayField from '../DisplayField/DisplayField';

class App extends Component {
  constructor() {
    super()
    this.state = {
      isLoading: true,
      isTab: 'Crawl',
      currentFilm: {},
      allCards: {
        people: [],
        planets: [],
        vehicles: [],
        favorites: []
      }
    }
  }

  componentDidMount() {
    const url = 'https://swapi.co/api/';
    let currState = this.state;

    findRandomFilm(`${url}films`)
      .then(singleFilm => this.setState({ currentFilm: singleFilm, isLoading: false }));
    
    let result;
    // findPeople(url, currState);
      let func = ()=>{
        result = findPeople(url, currState)
          .then(result => this.setState({result}))
        console.log('result', result)
      }

      func()
    
  }

  updateTab = (e) => {
    let tabName = e.target.id;
    this.setState({isTab: tabName})
  }

  render() {
    console.log(this.state)

    return (
      <div className="App">
        <div className="backgroundImg">
          <main>
            <Header updateTab={this.updateTab} />
            <DisplayField
              isLoading={this.state.isLoading}
              isTab={this.state.isTab}
              currentFilm={this.state.currentFilm}
              person= {this.state.allCards.people} />
          </main>
        </div>
      </div>
    );
  }
  
}

export default App;
