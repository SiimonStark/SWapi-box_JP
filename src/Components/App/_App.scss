html {
  background-color: black;
}

.App {
  position: relative;
  border: skyblue solid 5px;
  height: 100vh;
  width: 100vw;
  top: 0;
}

.backgroundImg {
  border: solid yellow 3px;
  width: 100vw;
  height: 100vh;
  display: grid;
  place-items: center;
  color: yellow;
  position: relative;
}

.backgroundImg::before {
  content: "";
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  position: absolute;
  background-image: url('https://wallpapertag.com/wallpaper/full/e/e/5/155073-amazing-star-wars-space-background-1980x1080-tablet.jpg');
  background-size: cover;
  filter: grayscale(100%);
}

main {
  position: relative;
  top: 0;
  height: 100%;
  width: 100%;
  display: grid;
  grid-template-areas: "Header" "DisplayField";
  grid-template-rows: 20% 80%; 
  border: magenta dotted 3px;
  font: bold 42px sans-serif;
}

header {
  grid-area: Header;
  box-sizing: border-box;
  position: relative;
  height: 100%;
}
.DisplayField {
  grid-area: DisplayField;
  box-sizing: border-box;
  height: 100%;
}