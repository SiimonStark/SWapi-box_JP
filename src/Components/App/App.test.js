import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import App from './App';
// import Header from '../Header/Header';
// import DisplayField from '../DisplayField/DisplayField';

describe('App', () => {
  let wrapper;
  beforeEach(()=>{
    wrapper = shallow(<App/>)
  });

  it('should match snapshot', ()=>{
    expect(wrapper).toMatchSnapshot();
  });

  it('has default state', ()=>{
    const expectedState = {
      isLoading: true,
      isTab: 'Crawl',
      currentFilm: {},
      allCards: {
        people: [],
        planets: [],
        vehicles: [],
        favorites: []
      }
    }
    expect(wrapper.state()).toEqual(expectedState);
  });

  it('should update isTab when button updateTab is called', ()=>{
    let mockEvent = {target: {id: 'People'}}

    wrapper.instance().updateTab(mockEvent);

    expect(wrapper.state('isTab')).toEqual('People');
  })
});
