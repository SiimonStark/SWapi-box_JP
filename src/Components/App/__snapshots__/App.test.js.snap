// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="backgroundImg"
  >
    <main>
      <Header
        updateTab={[Function]}
      />
      <DisplayField
        currentFilm={Object {}}
        isLoading={true}
        isTab="Crawl"
        person={Array []}
      />
    </main>
  </div>
</div>
`;
